### PR TITLE
Update API docs for BelongsToReference

### PR DESCRIPTION
**Pull request by bmac**
cc @pangratz

### CODE REVIEW

File: addon/-private/system/references/belongs-to.js [hunk 1]
```diff
@@ -5,6 +5,16 @@ import Reference from './reference';
 import isEnabled from 'ember-data/-private/features';
 import { assertPolymorphicType, deprecate } from "ember-data/-private/debug";
 
+
+/**
+   An BelongsToReference is a low level API that allows users and
```
Comment (bmac) [inline addon/-private/system/references/belongs-to.js]: "A BelongsToReference"

File: addon/-private/system/references/belongs-to.js [hunk 2]
```diff
@@ -26,19 +71,165 @@ BelongsToReference.prototype.remoteType = function() {
...
+        username: "@user"
+      }
+    }).then(function(user) {
+      userRef.value() === user;
+    });
+    ```
+
+   @method push
+   @param {Object|Promise} objectOrPromise a promise that resolves to a JSONAPI document object describing the new value of this relationship.
+   @return {Object} The meta information for the belongs-oo relationship.
```
Comment (bmac) [inline addon/-private/system/references/belongs-to.js]: This should say `push` returns the record.

File: addon/-private/system/references/belongs-to.js [hunk 3]
```diff
@@ -73,6 +308,42 @@ BelongsToReference.prototype.value = function() {
   return null;
 };
 
+/**
+   `value()` sycronously returns the current value of the belongs-to
+   relationship. Unlike `record.get('relationshipName')`, calling
+   `value()` on a reference does not trigger a fetch if the async
+   relationship is not yet loaded. If the relationship is not loaded
+   it will always return `null`. If the relationship is already loaded
+   this method does not trigger a new load.
```
Comment (bmac) [inline addon/-private/system/references/belongs-to.js]: This is copy and pasted from the value docs above.

File: addon/-private/system/references/belongs-to.js [hunk 4]
```diff
@@ -63,6 +253,50 @@ BelongsToReference.prototype.push = function(objectOrPromise) {
   });
 };
 
+/**
+   `value()` sycronously returns the current value of the belongs-to
```
Comment (pangratz) [inline addon/-private/system/references/belongs-to.js]: typo: `s/sycronously/synchronously/`

File: addon/-private/system/references/belongs-to.js [hunk 5]
```diff
@@ -26,19 +71,164 @@ BelongsToReference.prototype.remoteType = function() {
...
+
+   Example
+
+   ```javascript
+    // models/blog.js
+    export default DS.Model.extend({
+      user: DS.belongsTo({ async: true })
+    });
+
+    var blog = store.push({
```
Comment (pangratz) [inline addon/-private/system/references/belongs-to.js]: not a valid format for `store.push`

File: addon/-private/system/references/belongs-to.js [hunk 6]
```diff
@@ -26,19 +71,164 @@ BelongsToReference.prototype.remoteType = function() {
...
```
Comment (pangratz) [inline addon/-private/system/references/belongs-to.js]: not a valid format for `store.push`

File: addon/-private/system/references/belongs-to.js [hunk 7]
```diff
@@ -18,6 +28,41 @@ BelongsToReference.prototype = Object.create(Reference.prototype);
...
```
Comment (pangratz) [inline addon/-private/system/references/belongs-to.js]: not a valid format for `store.push`

File: addon/-private/system/references/belongs-to.js [hunk 8]
```diff
@@ -26,19 +71,164 @@ BelongsToReference.prototype.remoteType = function() {
...
+      relationships: {
+        user: {
+          data: { type: 'user', id: 1 }
+        }
+      }
+    });
+    var userRef = blog.belongsTo('user');
+
+    // provide data for reference
+    userRef.push({
```
Comment (pangratz) [inline addon/-private/system/references/belongs-to.js]: this needs to be a JSON-API doc of the form `{ data: { ... } }`

File: addon/-private/system/references/belongs-to.js [hunk 9]
```diff
@@ -63,6 +253,50 @@ BelongsToReference.prototype.push = function(objectOrPromise) {
...
+
+   Example
+
+    ```javascript
+    // models/blog.js
+    export default DS.Model.extend({
+      user: DS.belongsTo({ async: true })
+    });
+
+    var blog = store.push({
```
Comment (pangratz) [inline addon/-private/system/references/belongs-to.js]: not a valid format for `store.push`

File: addon/-private/system/references/belongs-to.js [hunk 10]
```diff
@@ -63,6 +253,50 @@ BelongsToReference.prototype.push = function(objectOrPromise) {
...
+          data: { type: 'user', id: 1 }
+        }
+      }
+    });
+    var userRef = blog.belongsTo('user');
+
+    userRef.value(); // null
+
+    // provide data for reference
+    userRef.push({
```
Comment (pangratz) [inline addon/-private/system/references/belongs-to.js]: this needs to be a JSON-API doc of the form `{ data: { ... } }`

File: addon/-private/system/references/belongs-to.js [hunk 11]
```diff
@@ -73,6 +307,39 @@ BelongsToReference.prototype.value = function() {
...
+
+   Example
+
+    ```javascript
+    // models/blog.js
+    export default DS.Model.extend({
+      user: DS.belongsTo({ async: true })
+    });
+
+    var blog = store.push({
```
Comment (pangratz) [inline addon/-private/system/references/belongs-to.js]: not a valid format for `store.push`

File: addon/-private/system/references/belongs-to.js [hunk 12]
```diff
@@ -85,6 +352,38 @@ BelongsToReference.prototype.load = function() {
...
```
Comment (pangratz) [inline addon/-private/system/references/belongs-to.js]: not a valid format for `store.push`

File: addon/-private/system/references/belongs-to.js [hunk 13]
```diff
@@ -26,19 +71,164 @@ BelongsToReference.prototype.remoteType = function() {
   return "id";
 };
 
+/**
+   The `id` of the record that this reference refers to. Together, the
+   `type()` and `id()` methods form a composite key for the identity
+   map. This can be used to access the id of an asycn relationship
```
Comment (pangratz) [inline addon/-private/system/references/belongs-to.js]: typo: `s/asycn/async/`

**Comment by bmac** [comment]
@pangratz I've updated this pr to address your comments.

**Comment by pangratz** [comment]
Thanks again @bmac!

**Pull request merged**